docs/SpaceTemplates.rst

```diff
@@ -0,0 +1,229 @@
+
+
+.. _PLFDevGuide.SpaceTemplates.Extend:
+
+============================
+Extending existing templates
+============================
+
+Sapce templates can be extended in order to add applications. Only applications can be added to space templates via extension.
+In order to add some applications to an existing template, ``extendSpaceTemplatePlugin`` method of the ``org.exoplatform.social.core.space.spi.SpaceTemplateService`` will be used in a component plugin.
+Add this to the configuration of your extension :
+
+    .. code:: xml
+
+        <external-component-plugins>
+            <target-component>org.exoplatform.social.core.space.spi.SpaceTemplateService</target-component>
+            <!-- Community space template -->
+            <component-plugin>
+              <name>Space Template Extension</name>
+              <set-method>extendSpaceTemplatePlugin</set-method>
+              <type>org.exoplatform.social.core.space.SpaceTemplateConfigPlugin</type>
+              <init-params>
+                <object-param>
+                  <name>template</name>
+                  <description>Space Template</description>
+                  <object type="org.exoplatform.social.core.space.SpaceTemplate">
+                    <field name="name"><string>community</string></field>
+                    <field name="applications">
+                      <collection type="java.util.ArrayList">
+                        <value>
+                          <object type="org.exoplatform.social.core.space.SpaceApplication">
+                            <field name="portletApp">
+                              <string>task-management</string>
+                            </field>
+                            <field name="portletName">
+                              <string>TaskManagementApplication</string>
+                            </field>
+                            <field name="appTitle">
+                              <string>abcd</string>
+                            </field>
+                            <field name="removable">
+                              <boolean>true</boolean>
+                            </field>
+                            <field name="order">
+                              <int>3</int>
+                            </field>
+                            <field name="uri">
+                              <string>tasks</string>
+                            </field>
+                          </object>
+                        </value>
+                      </collection>
+                    </field>
+                  </object>
+                </object-param>
+              </init-params>
+            </component-plugin>
+            ...
+        </external-component-plugins>
+
+This will add "tasks" application to the "community" space template. The "tasks" application will the third application on "community" template according to the "order" field.
```

Comment:
```suggestion
This will add "tasks" application to the "community" space template. The "tasks" application will be the third application in the "community" space created according to the "order" field.
```

Reply:
ok